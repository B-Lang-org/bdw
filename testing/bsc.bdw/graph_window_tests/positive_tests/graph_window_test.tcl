
# Opens the specified project
handlers::open_project project.bspec

# Displays the project status
commands::display_project_status

# Cleans the project
commands::full_clean

while {$BSPEC(FILES) == ""} {
        update
}

# Compiles the specified file considering dependencies 
handlers::compile_file mult4/Mult2.bsv -withdeps

while {$BSPEC(BUILDPID) != ""} {
        update
}

# Loads the specified module
handlers::load_module mkMult2

# Opens the specified graph window
handlers::show_graph -conflict 

# Opens the specified graph window
handlers::show_graph -exec

# Opens the specified graph window
handlers::show_graph -urgency 

# Opens the specified graph window
handlers::show_graph -combined 

# Opens the specified graph window
handlers::show_graph -combined_full 

# Minimizes all currently active windows except the Main window
commands::minimize_all

# Cleans the project
commands::full_clean

# Closes all currently opened windows except the Main window
commands::close_all

# Closes the project 
handlers::close_project

# Exits the workstation
exit
