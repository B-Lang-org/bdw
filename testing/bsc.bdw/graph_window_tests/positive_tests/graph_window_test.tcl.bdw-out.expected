COMP_BDIR = mult4
COMP_BSC_OPTIONS = -sched-dot
COMP_BSC_TYPE = bluesim
COMP_INFO_DIR = mult4
COMP_SIMDIR = mult4
COMP_TYPE = bsc
COPY_FLAGS = 0
COMP_VDIR = .
DIR = .
EDITOR_NAME = gvim
INCLUDED_FILES = *.bsv
LINK_BSC_OPTIONS = -keep-fires
LINK_OUTDIR = .
LINK_OUTNAME = out
LINK_TYPE = bsc
MAIN_PLACEMENT = 700x350+0+0
MODULE_PLACEMENT = 550x350+50+500
NAME = project
PATHS = mult4
PATHS = .
PATHS = Libraries
PACKAGE_PLACEMENT = 770x420+475+150
PROJECT_PLACEMENT = 250x460+0+385
SCHEDULE_PLACEMENT = 650x650+500+225
SIM_NAME = iverilog
STATUS = saved
TOP_FILE = mult4/Mult4Tb.bsv
TOP_MODULE = mkMult4Tb
TYPE_PLACEMENT = 770x420+475+540
VIEWER_CLOSE = close
WIZARD_PLACEMENT = 1024x700+50+50
EDITOR_GVIM = gvim
EDITOR_EMACS = emacs
EDITOR_OTHER = gedit +%n %f
Executing: bsc -u -sim -simdir mult4 -bdir mult4 -info-dir mult4 -sched-dot -p .:%/Libraries  mult4/Mult2.bsv
WARNING: + bsc -u -sim -simdir mult4 -bdir mult4 -info-dir mult4 -sched-dot -p .:%/Libraries mult4/Mult2.bsv
RESULT: checking package dependencies
RESULT: compiling mult4/Mult.bsv
RESULT: compiling mult4/Mult2.bsv
RESULT: code generation for mkMult2 starts
RESULT: DOT file created: mult4/mkMult2_conflict.dot
RESULT: DOT file created: mult4/mkMult2_exec.dot
RESULT: DOT file created: mult4/mkMult2_urgency.dot
WARNING: Warning: "mult4/Mult2.bsv", line 23, column 8: (G0117)
WARNING:   Rule `start' shadows the effects of `cycle' when they execute in the same
WARNING:   clock cycle. Affected method calls:
WARNING:     product.write, mcand.write, mplr.write
WARNING:   To silence this warning, use the `-no-warn-action-shadowing' flag.
RESULT: DOT file created: mult4/mkMult2_combined.dot
RESULT: DOT file created: mult4/mkMult2_combined_full.dot
RESULT: Elaborated module file created: mult4/mkMult2.ba
RESULT: All packages are up to date.
mkMult2
The Conflict Graph is opened.
The Execution Graph is opened.
The Urgency Graph is opened.
The Combined Graph is opened.
The Combined Full Graph is opened.
